### ISSUE TITLE
[BUG]  Stripe payment dialogue not moving when virtual keyboard appears.

### ISSUE THREAD

**Issue by Mattheisen** [stripe/stripe-ios#2495]
## Summary
 When the stripe dialogue about the payment starts, it appears from the bottom of the app screen. When a user taps on an input field of this form, the virtual keyboard of the iPhone will appear and then the entire stripe dialogue moves to the upper part of the screen. This constellation is fine and happening in about 80% of the cases. However, in the other 20%, the stripe dialogue does not move to the top of the screen and remains in its bottom position. This bug will cause that the dialogue is overlapped by the appearing virtual keyboard. Hence, the user cannot see what he is typing into the stripe dialogue field.

## Code to reproduce
. Here is the section of our code in which we use your SDK. However, as mentioned above, it is not reproducible and happens only occasionally. So this provided code will not necessarily show the phenomena. 
            // Initialize the payments SDK
            STPAPIClient.shared.publishableKey = answerFromPaymentStripeBackend.publishableKey
            var configuration = PaymentSheet.Configuration()
            configuration.merchantDisplayName = "OneBrush Gmbh"
            configuration.customer = .init(id: answerFromPaymentStripeBackend.customer, ephemeralKeySecret: answerFromPaymentStripeBackend.ephemeralKey)
            configuration.returnURL = "dentinostic://stripe-redirect"
            configuration.defaultBillingDetails.address.country = "DE"
            configuration.allowsDelayedPaymentMethods = false
            let showApplePay = true
            if showApplePay {
                configuration.applePay = .init(merchantId: "merchant.de.onebrush.dentinostic", merchantCountryCode: "DE")
            }
                let paymentSheet = PaymentSheet(paymentIntentClientSecret:   answerFromPaymentStripeBackend.paymentIntent,
                                                configuration: configuration)
                paymentSheet.present(from: self) { paymentResult in
                    
                    // The user has completed interacting with the payment UI,
                    // get the result and call our completion callback
                    let status: PaymentStatus
                    var myPaymentResult = ""
                    switch paymentResult {
                    case .completed:
                        myPaymentResult = "success"
                    case .canceled:
                        myPaymentResult = "cancelled"
                    case .failed(let error):
                        myPaymentResult = "failure"
                    }
                }

## iOS version

Can happen on all version from 15.2 upwards

## Installation method

with cocopods

## SDK version

Here is the relevant podfile.lock section:
- Stripe (22.8.4):
    - Stripe/Stripe3DS2 (= 22.8.4)
    - StripeApplePay (= 22.8.4)
    - StripeCore (= 22.8.4)
    - StripeUICore (= 22.8.4)
  - Stripe/Stripe3DS2 (22.8.4):
    - StripeApplePay (= 22.8.4)
    - StripeCore (= 22.8.4)
    - StripeUICore (= 22.8.4)
  - StripeApplePay (22.8.4):
    - StripeCore (= 22.8.4)
  - StripeCore (22.8.4)
  - StripeUICore (22.8.4):
    - StripeCore (= 22.8.4)

## Other information

Is there any option to pin the dialogue to the top of the screen area? We could live with it when it is positioned in the upper part generally until the bug is resolved.

**Comment by yuki-stripe** [comment]
Hello @Mattheisen, thank you for the bug report.  Could you provide a video of the behavior and include your Xcode version, whether it's a device or simulator, and the iOS version?

Also, can you reproduce this issue on a device?

**Comment by Mattheisen** [comment]
Hello Yuki, sorry for the late reply. In the meantime we had continued with our investigations and found a solution. At first we found how we could reproduce this error: It always happened when the screen of our app got locked (programmatically) and a PIN entry is required to open the app again. Whenever this happened before the stripe dialogue was started, the problem with the dialogue screen position as described above appeared. Then we found that a change of the modalPresentationStyle of this screenLock ViewController impacted the behaviour of the later started stripe dialogue. We changed this parameter from .fullScreen to .overFullScreen. This solved it.
